### PR TITLE
Handle escaped uris for creating AppInfo in FileManager1.vala

### LINKED ISSUE #42: Opening paths containing spaces via org.freedesktop.FileManager1 fails.
A properly URI encoded path such as the following sent to the `ShowItems` method causes Files to try and open an incorrect path.
`file:///home/david/Downloads/Torrent%20Downloads/podcast.mp3`

For the example above Files tries and fails to open the following path:
`/home/david/Downloads/Torrent0Downloads/podcast.mp3/`

I have tested with Nautilus to check that the path I'm sending is well formed and it had no issue.

### PR DESCRIPTION
**Pull request by jeremypw**
Fixes #42 

To test:

After installing and killing any existing Files daemon, open a terminal and enter:

```
dbus-send --session --print-reply --reply-timeout=120000 --type=method_call \
--dest='org.freedesktop.FileManager1'   '/org/freedesktop/FileManager1' \
org.freedesktop.FileManager1.ShowFolders   array:string:"[path to be tested]" string:"TESTID"
```
Replace `[path to be tested]` with the path to an existing file or folder with spaces in the name.  The spaces may or may not be escaped (%20).

The branch should also cope with other unusual characters in the name although it cannot be guaranteed that all weird paths will be handled properly.

### CODE REVIEW

File: pantheon-files-daemon/FileManager1.vala [hunk 1]
```diff
@@ -39,29 +40,28 @@ public class FileManager1 : Object {
...
 
-        if (pf_app_info != null) {
-            try {
-                pf_app_info.launch (null, null);
-            } catch (Error e) {
-                var msg = "Unable to open item or folder with command %s. %s".printf (cmd, e.message);
-                throw new IOError.FAILED (msg);
-            }
-        }
+    private string prepare_uri_for_appinfo_create (string uri, bool allow_utf8 = true) {
```
Comment (donadigo) [inline pantheon-files-daemon/FileManager1.vala]: This should be static.

File: pantheon-files-daemon/FileManager1.vala [hunk 2]
```diff
@@ -18,6 +18,7 @@
 
 [DBus (name = "org.freedesktop.FileManager1")]
 public class FileManager1 : Object {
+    const string reserved_chars = (GLib.Uri.RESERVED_CHARS_GENERIC_DELIMITERS + GLib.Uri.RESERVED_CHARS_SUBCOMPONENT_DELIMITERS);
```
Comment (donadigo) [inline pantheon-files-daemon/FileManager1.vala]: `reserved_chars` should be uppercase: `RESERVED_CHARS` for constants.

**Comment by jeremypw** [comment]
Since we are having to fix at least two reports we might as well fix as many corner cases as possible, so long as it does not significantly complicate the code.  There is a custom RESERVED_CHAR string already in the main code but the daemon does not have access to that.

File: pantheon-files-daemon/FileManager1.vala (hunk 2 above)
Comment (jeremypw) [reply pantheon-files-daemon/FileManager1.vala]: modified to static string with replacements already in place.

**Comment by jeremypw** [comment]
I have tested this with a (legal) filename containing spaces, Japanese characters, and punctuation including %.

**Pull request merged**